src/components/NcRichText/autolink.js

```diff
@@ -84,20 +84,37 @@ export const parseUrl = (text) => {
 	return text
 }
 
-export const getRoute = (router, url) => {
+/**
+ * Try to get mapped path from router's routes
+ *
+ * @param {import('vue-router').default} router - VueRouter instance
+ * @param {string} urlString - URL to parse
+ * @return {string|null} path (RouteLocationPathRaw) for router link or null if path does not match any route
+ */
+export const getRoute = (router, urlString) => {
 	// Skip if Router is not defined in app, or baseUrl does not match
-	if (!router || !url.includes(getBaseUrl())) {
+	if (!router || !urlString.startsWith(getBaseUrl())) {
 		return null
 	}
 
-	const regexArray = router.getRoutes()
-		// route.regex matches only complete string (^.$), need to remove these characters
-		.map(route => new RegExp(route.regex.source.slice(1, -1), route.regex.flags))
+	try {
+		const url = new URL(urlString)
+		// Remove optional index.php from path
+		const path = url.pathname.replace(/^\/index\.php$/, '')
```

Comment:
```suggestion
		const path = url.pathname.replace(/^\/index\.php$/, '') + url.hash
```

Works for me as expected =)

Reply:
Outdated, an approach has much changed